environments/se/core/src/main/java/org/jboss/weld/environment/se/contexts/activator/RequestScopeActiveInterceptor.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2016, Red Hat, Inc. and/or its affiliates, and individual
+ * contributors by the @authors tag. See the copyright.txt in the
+ * distribution for a full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.weld.environment.se.contexts.activator;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.Priority;
+import javax.interceptor.AroundInvoke;
+import javax.interceptor.Interceptor;
+import javax.interceptor.InvocationContext;
+
+import org.jboss.weld.Container;
+import org.jboss.weld.bootstrap.api.helpers.RegistrySingletonProvider;
+import org.jboss.weld.context.bound.BoundRequestContext;
+import org.jboss.weld.manager.BeanManagerImpl;
+
+@Interceptor
+@RequestScopeActive
+@SuppressWarnings("checkstyle:magicnumber")
+@Priority(Interceptor.Priority.LIBRARY_BEFORE + 100)
+public class RequestScopeActiveInterceptor {
+
+    private BeanManagerImpl beanManager;
+
+    @AroundInvoke
+    Object invoke(InvocationContext invocationContext) throws Exception {
+        if (beanManager == null) {
+            beanManager = obtainBeanManager(invocationContext);
+        }
+        BoundRequestContext boundRequestContext = beanManager.instance().select(BoundRequestContext.class).get();
+
+        if (!boundRequestContext.isActive()) {
+            Map<String, Object> storage = new HashMap<>();
+            boundRequestContext.associate(storage);
+            boundRequestContext.activate();
+            Object obj = invocationContext.proceed();
```

Comment:
We should probably use `finally` block here and always invalidate, deactive, etc.